scripts/e2e.npm.publish.sh

```diff
@@ -34,11 +34,30 @@ npx npm-auth-to-token \
   -e test@test.com \
   -r http://localhost:4873
 
-# `npm version patch`
-npx lerna exec -- npm version patch
+# Prep branch for Lerna's git-checks
+BRANCH=$TRAVIS_PULL_REQUEST_BRANCH
+if [ -z "$BRANCH" ]; then
 
-# `npm publish`
-npx lerna exec --concurrency 1 -- npm publish \
-  --tag e2e \
+  BRANCH=$TRAVIS_BRANCH
+
+fi
+
+git checkout $BRANCH --
```

Comment:
The build is failing here because 
+ Lerna [won't publish to the virtual registry][1] when the git head is in a detached state. 
+ It's necessary to explicitly checkout a branch to trick it
+ The branch isn't present when a fork's changes are proposed for the upstream

I'm going to resolve this temporarily by merging into a staging branch at web3 to make sure the CI runs as expected here. It would be nice if this ran for outside contributors, but perhaps is not critical that it does either because this kind of test is more of a pre-publication tests that's use internally anyway. 

https://github.com/lerna/lerna/issues/1595#issuecomment-422532097